langchain4j-azure-ai-search/src/main/java/dev/langchain4j/store/embedding/azure/search/AbstractAzureAiSearchEmbeddingStore.java

```diff
@@ -68,8 +68,15 @@ protected void initialize(String endpoint, AzureKeyCredential keyCredential, Tok
             // if the indexName is provided, it will be used when creating the default index
             throw new IllegalArgumentException("index and indexName cannot be both defined");
         }
+        if (createOrUpdateIndex && index != null) {
```

Comment:
Is it really what you want ?
I though it should be `(createOrUpdateIndex && index == null)`, because how do you create or update an index programmaticaly without the description of the index ?